test/utils.js

```diff
@@ -7,8 +7,42 @@ import { Provider } from 'react-redux';
 import thunk from 'redux-thunk';
 import sinon from 'sinon';
 
+import _get from 'lodash/get';
+import toPath from 'lodash/toPath';
+import i from 'icepick';
+import Immutable from 'immutable';
+
 import { controls, modelReducer, formReducer, Control, actions } from '../src';
 
+export const defaultTestContexts = {
+  standard: {
+    object: {},
+    get: (value, path) => {
+      if (!path) return value;
+      return _get(value, path);
+    },
+    set: (state, path, value) => i.setIn(state, path, value),
+    length: (value) => value.length,
+    getInitialState: (state) => state,
+  },
+  immutable: {
+    object: new Immutable.Map(),
+    get: (value, path) => {
+      if (!path) return value.toJS();
+
+      const result = value.getIn(toPath(path));
+      try {
+        return result.toJS();
+      } catch (e) {
+        return result;
+      }
+    },
```

Comment:
I also made some changes to the get functions in here to take a value without a path as I was finding that I needed to do that a number of times in form-component-spec.js.  So far the rest of the specs don't have that change, so if you like this approach it will help to convert all those over to using the default from here.